ossl/src/lib.rs

```diff
@@ -239,10 +244,21 @@ impl OsslContext {
         }
     }
 
-    pub fn load_default_configuration(&self) -> Result<(), Error> {
+    pub fn load_default_configuration(&mut self) -> Result<(), Error> {
         self.load_configuration_file(None)
     }
 
+    pub fn load_legacy_provider(&mut self) -> Result<(), Error> {
+        let provider =
+            unsafe { OSSL_PROVIDER_load(self.ptr(), c"legacy".as_ptr()) };
```

Comment:
should we do the load unconditionally without checking if it was not loaded through the configuration file or something? I am doing the following with the pkcs11 provider in the libssh:
https://gitlab.com/libssh/libssh-mirror/-/blob/master/src/libcrypto.c#L105